lib/assert.js

```diff
@@ -80,6 +84,55 @@ const assert = module.exports = ok;
 
 const NO_EXCEPTION_SENTINEL = {};
 
+/**
+ * Assert options.
+ * @typedef {object} AssertOptions
+ * @property {'full'|'simple'} [diff='simple'] - If set to 'full', shows the full diff in assertion errors.
+ * @property {boolean} [strict=true] - If set to true, non-strict methods behave like their corresponding
+ *   strict methods.
+ */
+
+/**
+ * @class Assert
+ * @param {AssertOptions} [options] - Optional configuration for assertions.
+ * @throws {ERR_INVALID_ARG_TYPE} If not called with `new`.
+ */
+function Assert(options) {
+  if (!new.target) {
+    throw new ERR_INVALID_ARG_TYPE('Assert', 'constructor', Assert);
```

Comment:
Can we just use class syntax here?

Reply:
Can we use `ERR_CONSTRUCT_CALL_REQUIRED` instead of `ERR_INVALID_ARG_TYPE`? That seems much more fitted

https://github.com/nodejs/node/blob/5f7dbf45a3d3e3070d5f58f9a9c2c43dbecc8672/doc/api/errors.md#L816-L818

Reply:
Actually, found that some error types were missing [59022](https://github.com/nodejs/node/pull/59022)
I'll keep it like it is for now